bare_header/sifive_l2hwpf0.h

```diff
@@ -0,0 +1,40 @@
+/* Copyright 2019 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef __BARE_HEADER_SIFIVE_L2HWPF0__H
+#define __BARE_HEADER_SIFIVE_L2HWPF0__H
+
+#include "bare_header/device.h"
+
+#include <regex>
+
+class sifive_l2hwpf0 : public Device {
+public:
+  sifive_l2hwpf0(std::ostream &os, const fdt &dtb)
+      : Device(os, dtb, "sifive,spf0") {}
+
+  int get_index(const node &n) { return Device::get_index(n, compat_string); }
+
+  void emit_defines() {
+    dtb.match(std::regex(compat_string), [&](node n) {
+      emit_comment(n);
+
+      emit_base(n);
+      emit_size(n);
+
+      os << std::endl;
+    });
+  }
+
+  void emit_offsets() {
+    if (dtb.match(std::regex(compat_string), [](const node n) {}) != 0) {
+      emit_compat();
+      emit_offset("REG0", 0x00);
+      emit_offset("REG1", 0x04);
+
+      os << std::endl;
+    }
+  }
```

Comment:
We also need these definitions for these values for init
        sifive,spf-queue-entries = <16>;
        sifive,spf-window-bits = <6>;
        sifive,spf-distance-bits = <6>;
        sifive,spf-streams = <8>;